src/repeated_game.jl

```diff
@@ -0,0 +1,367 @@
+    b = Array(Float64, nH + 2)
+
+    return c, A, b
+end
+
+"""
+Given a constraint w ∈ W, this finds the worst possible payoff for agent i
+
+The output of this function is used to create the values associated with
+incentive compatibility constraints
+
+The arguments for this function are
+  * rpd: Two player repeated game
+  * H: Subgradients used to approximate value set
+  * C: Hyperplane levels for value set approximation
+  * i: Which player want worst value for
+"""
+function worst_value_i(rpd::RepGame2, H::Array{Float64, 2}, C::Array{Float64, 1}, i::Int)
+    # Objective depends on which player we are minimizing
+    c = zeros(2)
+    c[i] = 1.0
+
+    # Lower and upper bounds for w
+    lb = [-Inf, -Inf]
+    ub = [Inf, Inf]
+
+    lpout = linprog(c, H, '<', C, lb, ub, ClpSolver())
+    if lpout.status == :Optimal
+        out = lpout.sol[i]
+    else
+        out = minimum(rpd.sg.players[i].payoff_array)
+    end
+
+    return out
+end
+
+"See worst_value_i for documentation"
+worst_value_1(rpd::RepGame2, H::Array{Float64, 2}, C::Array{Float64, 1}) =
+    worst_value_i(rpd, H, C, 1)
+"See worst_value_i for documentation"
+worst_value_2(rpd::RepGame2, H::Array{Float64, 2}, C::Array{Float64, 1}) =
+    worst_value_i(rpd, H, C, 2)
+"See worst_value_i for documentation"
+worst_values(rpd::RepGame2, H::Array{Float64, 2}, C::Array{Float64, 1}) =
+    (worst_value_1(rpd, H, C), worst_value_2(rpd, H, C))
+
+#
+# Outer Hyper Plane Approximation
+#
+"""
+Approximates the set of equilibrium value set for a repeated game with the
+outer hyperplane approximation described by Judd, Yeltekin, Conklin 2002
+
+NOTE: If your code fails then it might be the case that the value set is
+      only the value which corresponds to the pure action nash equilibrium
+      or you might just need more points to be precise enough for the
+      Polyhedra library to be able to convert to the vertice representation.
+
+The arguments are
+  * rpd: 2 player repeated game
+
+The keyword arguments are
+  * nH: Number of subgradients used in approximation
+  * tol: Tolerance in differences of set
+  * maxiter: Maximum number of iterations
+  * verbose: Whether to display updates about iterations and distance
+  * nskipprint: Number of iterations between printing information (verbose=true)
+  * pane_check: Whether to perform a check about whether a pure action ne exists
+
+"""
+function outerapproximation(rpd::RepGame2; nH=32, tol=1e-8, maxiter=500,
+                            verbose=false, nskipprint=50)
+    # Long unpacking of stuff
+    sg, delta = unpack(rpd)
+    p1, p2 = sg.players
+    po_1, po_2 = p1.payoff_array, p2.payoff_array
+    p1_minpayoff, p1_maxpayoff = extrema(po_1)
+    p2_minpayoff, p2_maxpayoff = extrema(po_2)
+
+    # Check to see whether at least one pure strategy NE exists
+    pane = pure_nash(sg)
+    no_pane_exists = (length(pane) == 0)
+    if no_pane_exists
+        error("No pure action Nash equilibrium exists in stage game")
```

Comment:
Given that `min_pane_payoffs` will no longer be used:

* Would it make sense to write a generator version (if any such exists in Julia) of `pure_nash` and then here go forward as soon as *one* pure action Nash equilibrium is found? There will be no difference in the worse case, but it should take a strictly smaller amount of time "on average".

* I think it is better to keep the option to skip this part, as was added at one of the previous commits.